R/Canada.R

```diff
@@ -24,10 +24,46 @@ Canada <- R6::R6Class("Canada",
     level_1_region = "level_1_region", # for brevity - refers to
     # provinces and territories
     #' @field data_url link to raw data
-    data_url = "https://health-infobase.canada.ca/src/data/covidLive/covid19.csv", #nolint
+    data_url = "https://health-infobase.canada.ca/src/data/covidLive/covid19.csv", # nolint
     #' @field source_data_cols existing columns within the raw data
-    source_data_cols = c("cases_new", "cases_total", "deaths_new",
-      "recovered_total", "tested_new"),
+    source_data_cols = c(
+      "cases_new", "cases_total", "deaths_new",
+      "recovered_total", "tested_new"
+    ),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    set_region_codes = function() {
+      message_verbose(
+        self$verbose,
+        paste(
+          "Getting region codes for",
+          self$country
+        )
+      )
+      level_1_canada <- tibble::tibble(
+        level_1_region_code = c(
+          "CA-AB", "CA-BC", "CA-MB", "CA-NB", "CA-NL",
+          "CA-NS", "CA-NT", "CA-NU", "CA-ON", "CA-PE",
+          "CA-QC", "CA-SK", "CA-YT"
+        ),
+        region_level_1 = c(
+          "Alberta", "British Columbia", "Manitoba", "New Brunswick",
+          "Newfoundland and Labrador", "Nova Scotia", "Northwest Territories",
+          "Nunavut", "Ontario", "Prince Edward Island", "Quebec",
+          "Saskatchewan", "Yukon"
+        )
+      )
+      canada_codes <- tibble::tibble(
+        country = "canada",
+        level = c("level_1_region"),
+        name = c("iso_3166_2"),
+        codes = list(
+          level_1_canada
```

Comment:
only this needs to be assigned in this function and in fact maybe doesn't even need to be if it can be built from a field?